planning/behavior_path_planner/src/scene_module/pull_over/util.cpp

```diff
@@ -344,6 +68,34 @@ lanelet::ConstLanelets getPullOverLanes(const RouteHandler & route_handler)
   return pull_over_lanes;
 }
 
+bool hasEnoughDistanceToParkingStart(
+  const PathWithLaneId & path, const Pose & current_pose, const Pose & start_pose,
+  const double current_vel, const double maximum_deceleration, const double decide_path_distance,
+  const double ego_nearest_dist_threshold, const double ego_nearest_yaw_threshold)
+{
+  const size_t ego_segment_idx = motion_utils::findFirstNearestSegmentIndexWithSoftConstraints(
+    path.points, current_pose, ego_nearest_dist_threshold, ego_nearest_yaw_threshold);
+  const size_t start_segment_idx = motion_utils::findFirstNearestSegmentIndexWithSoftConstraints(
+    path.points, start_pose, ego_nearest_dist_threshold, ego_nearest_yaw_threshold);
```

Comment:
Lemme check.
I'm not sure what the `start_pose` is, but is it ok to use find"First"NearestSegmentIndexWithSoftConstraints?

Reply:
`start_pose` is start pull over pose of ego. I think it's ok to use this function same to ego_pose.